application/gqlgen/gqlgenutils.go

```diff
@@ -37,13 +38,34 @@ func convertGqlLocationInputToDBLocation(input LocationInput) models.Location {
 	return l
 }
 
-func convertUserPreferencesToKeyValues(input []UpdateUserPreferenceInput) [][2]string {
-	keyVals := make([][2]string, len(input))
-	for i, p := range input {
-		keyVals[i] = [2]string{p.Key, p.Value}
+func convertUserPreferencesToStandardPreferences(input *UpdateUserPreferencesInput) (models.StandardPreferences, error) {
+	if input == nil {
+		return models.StandardPreferences{}, nil
 	}
 
-	return keyVals
+	stPrefs := models.StandardPreferences{}
+
+	if input.Language != nil {
+		lang := strings.ToLower(fmt.Sprintf("%v", *input.Language))
+		if !domain.IsLanguageAllowed(lang) {
+			return models.StandardPreferences{}, errors.New("user preference language not allowed ... " + lang)
+		}
+		stPrefs.Language = lang
+	}
+
+	if input.TimeZone != nil {
+		stPrefs.TimeZone = strings.ToLower(*input.TimeZone)
```

Comment:
We'll need some validation of the timezone

Reply:
Done